### DISCUSSION TITLE
Getting this error : Error while executing command: gauge run specs. Reason: exit status 1... during Gauge python build on AWS

### DISCUSSION THREAD

**Discussion by premnhs** [getgauge/gauge#1921] [General]
Hi

I m getting the following error, while building a gauge python project on aws build, I dint had this error before, this suddenly started to occur.

Not sure what is causing this. Any pointers plz..thanks

```
Successfully installed baron-0.9 debugpy-1.2.1 getgauge-0.3.14 grpcio-1.34.0 protobuf-3.15.3 redBaron-0.9.2 rply-0.7.8
229 | WARNING: You are using pip version 20.3.3; however, version 21.0.1 is available.
230 | You should consider upgrading via the '/root/.pyenv/versions/3.9.1/bin/python3.9 -m pip install --upgrade pip' command.
231 |  
232 | [Container] 2021/02/26 12:48:49 Running command cd test/IntegrationTests
233 |  
234 | [Container] 2021/02/26 12:48:49 Running command gauge run specs
235 | Compatible version of plugin html-report not found. Installing plugin html-report...
236 | ........................
237 | Successfully installed plugin 'html-report' version 4.0.12
238 | Installing required plugins.
239 | .....................
240 | Successfully installed plugin 'screenshot' version 0.0.1
241 | Python: 3.9.1
242 | Error occurred while waiting for runner process to finish.
243 | Error : exit status 1
244 |  
245 | Error Message: transport is closing
246 | Stacktrace:
247 |  
248 |  
249 | Error Message: Runner is not Alive
250 | Stacktrace:
251 |  
252 | Successfully generated html-report to => /IntegrationTests/reports/html-report/index.html
253 |  
254 | Failed to kill Runner: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:43151: connect: connection refused"
255 | Specifications: 0 executed  0 passed    0 failed    0 skipped
256 | Scenarios:  0 executed  0 passed    0 failed    0 skipped
257 |  
258 | Total time taken: 373ms
259 |  
260 | [Container] 2021/02/26 12:48:54 Command did not exit successfully gauge run specs exit status 1
261 | [Container] 2021/02/26 12:48:54 Phase complete: POST_BUILD State: FAILED
262 | [Container] 2021/02/26 12:48:54 Phase context status code: COMMAND_EXECUTION_ERROR Message: Error while executing command: gauge run specs. Reason: exit status 1
263

```

**Comment by premnhs** [comment]
> Hi
> 
> I m getting the following error, while building a gauge python project on aws build, I dint had this error before, this suddenly started to occur.
> 
> Not sure what is causing this. Any pointers plz..thanks
> 
> ```
> Successfully installed baron-0.9 debugpy-1.2.1 getgauge-0.3.14 grpcio-1.34.0 protobuf-3.15.3 redBaron-0.9.2 rply-0.7.8
> 229 | WARNING: You are using pip version 20.3.3; however, version 21.0.1 is available.
> 230 | You should consider upgrading via the '/root/.pyenv/versions/3.9.1/bin/python3.9 -m pip install --upgrade pip' command.
> 231 |  
> 232 | [Container] 2021/02/26 12:48:49 Running command cd test/IntegrationTests
> 233 |  
> 234 | [Container] 2021/02/26 12:48:49 Running command gauge run specs
> 235 | Compatible version of plugin html-report not found. Installing plugin html-report...
> 236 | ........................
> 237 | Successfully installed plugin 'html-report' version 4.0.12
> 238 | Installing required plugins.
> 239 | .....................
> 240 | Successfully installed plugin 'screenshot' version 0.0.1
> 241 | Python: 3.9.1
> 242 | Error occurred while waiting for runner process to finish.
> 243 | Error : exit status 1
> 244 |  
> 245 | Error Message: transport is closing
> 246 | Stacktrace:
> 247 |  
> 248 |  
> 249 | Error Message: Runner is not Alive
> 250 | Stacktrace:
> 251 |  
> 252 | Successfully generated html-report to => /IntegrationTests/reports/html-report/index.html
> 253 |  
> 254 | Failed to kill Runner: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:43151: connect: connection refused"
> 255 | Specifications: 0 executed  0 passed    0 failed    0 skipped
> 256 | Scenarios:  0 executed  0 passed    0 failed    0 skipped
> 257 |  
> 258 | Total time taken: 373ms
> 259 |  
> 260 | [Container] 2021/02/26 12:48:54 Command did not exit successfully gauge run specs exit status 1
> 261 | [Container] 2021/02/26 12:48:54 Phase complete: POST_BUILD State: FAILED
> 262 | [Container] 2021/02/26 12:48:54 Phase context status code: COMMAND_EXECUTION_ERROR Message: Error while executing command: gauge run specs. Reason: exit status 1
> 263
> ```

03/03 : Update on this :   

Appears to be its the permission issue when performing a outbound call,
Noticed on  the code that when running the test, we had to use : 

`dev = boto3.session.Session(profile_name='local profile name ')`



but

when run on build, we don't need to provide this.  So sorted it by creating different env properties.

This can be closed.

**Reply by sriv** [reply]
Thanks for sharing your findings.
